### PR TITLE
fix: optimize mempool transaction reads and writes

### PR DESCRIPTION
**Pull request by rafaelcr**
* Delete `mempool_digest` materialized view, keep data in `chain_tip` table
* Change the digest algorithm from a hash to a simple "updated at" timestamp so mempool writes are extremely cheap
* Keep `tx_count` column to avoid running ad-hoc counts over unpruned txs
* Insert mempool txs in batches

### CODE REVIEW

File: src/datastore/pg-write-store.ts [hunk 1]
```diff
@@ -2747,28 +2761,6 @@ export class PgWriteStore extends PgStore {
...
-   * Called when a full event import is complete.
-   */
-  async finishEventReplay() {
-    if (!this.isEventReplay) return;
-    await this.refreshMaterializedView('mempool_digest', this.sql, false);
-  }
-
   /**
    * batch operations (mainly for event-replay)
    */
```
[the remark this answers is not in the document]
Comment (rafaelcr) [reply src/datastore/pg-write-store.ts:2774]: Great catch, done!

File: src/tests/mempool-tests.ts [hunk 2]
```diff
@@ -1546,7 +1546,7 @@ describe('mempool tests', () => {
     // directly inserting the mempool-tx and mined-tx, bypassing the normal update functions.
     await db.updateBlock(db.sql, dbBlock1);
     const chainTip = await db.getChainTip();
-    await db.insertDbMempoolTx(mempoolTx, chainTip, db.sql);
+    await db.insertDbMempoolTxs([mempoolTx], chainTip, db.sql);
     await db.updateTx(db.sql, dbTx1);
 
     // Verify tx shows up in mempool (non-pruned)
```
Comment (zone117x) [inline src/tests/mempool-tests.ts:1552]: Would it be reasonable to check for the new behavior in the `Reconcile mempool pruned status` test? Not sure if that makes sense, up to you

**Review by zone117x — APPROVED** [review]
Left a comment about adding another test, otherwise LGTM!

File: src/tests/mempool-tests.ts (hunk 2 above)
Comment (rafaelcr) [reply src/tests/mempool-tests.ts:1552]: Good idea, done

**Pull request merged**